File: print-manager-dotnet-core-cs/PageScalingExample.cs
using System;
using ceTe.DynamicPDF.Printing;

namespace print_manager_dotnet_core_cs
{
    class PageScalingExample
    {
        public static void RunActualPaperScaling()
        {
            InputPdf pdf = new InputPdf(Util.GetPath("Resources/DocumentA.pdf"));

            // Create a print job with the document to be printed to the default printer
            PrintJob printJob = new PrintJob(Printer.Default);
            printJob.Pages.Add(pdf);
            printJob.PrintOptions.Scaling = PageScaling.ActualSize;

            //Print the job
            printJob.Print();
        }

        public static void RunMultiPageScaling()
        {
            InputPdf pdf = new InputPdf(Util.GetPath("Resources/DocumentA.pdf"));

            // Create a print job with the document to be printed to the default printer
            PrintJob printJob = new PrintJob(Printer.Default);
            printJob.Pages.Add(pdf);

            // Set MultipagePageScaling
            printJob.PrintOptions.Scaling = new MultipagePageScaling(4, 2);

            //Print the job
            printJob.Print();
        }

        public static void RunPageEdgeScaling()
        {
            InputPdf pdf = new InputPdf(Util.GetPath("Resources/DocumentA.pdf"));

            // Create a print job with the document to be printed to the default printer
            PrintJob printJob = new PrintJob(Printer.Default);
            printJob.Pages.Add(pdf);

            // Set AutoPageScaling
            printJob.PrintOptions.Scaling = new AutoPageScaling(ScaleTo.PageEdges);

            //Print the job
            printJob.Print();
        }

        public static void RunPercentageScaling()
        {
            InputPdf pdf = new InputPdf(Util.GetPath("Resources/DocumentA.pdf"));

            // Create a print job with the document to be printed to the default printer
            PrintJob printJob = new PrintJob(Printer.Default);
            printJob.Pages.Add(pdf);

            // Set PercentagePageScaling
            printJob.PrintOptions.Scaling = new PercentagePageScaling(0.75f);

            //Print the job
            printJob.Print();
        }
    }
}

File: print-manager-dotnet-core-vb/PageScalingExample.vb
Imports ceTe.DynamicPDF.Printing

Public Class PageScalingExample
    Public Shared Sub RunActualPaperScaling()
        Dim pdf As InputPdf = New InputPdf(Util.GetPath("Resources/DocumentA.pdf"))

        ' Create a print job with the document to be printed to the default printer
        Dim MyPrintJob As PrintJob = New PrintJob(Printer.Default)
        MyPrintJob.Pages.Add(pdf)
        MyPrintJob.PrintOptions.Scaling = PageScaling.ActualSize

        ' Print the print job
        MyPrintJob.Print()
    End Sub

    Public Shared Sub RunMultiPageScaling()
        Dim pdf As InputPdf = New InputPdf(Util.GetPath("Resources/DocumentA.pdf"))

        ' Create a print job with the document to be printed to the default printer
        Dim MyPrintJob As PrintJob = New PrintJob(Printer.Default)
        MyPrintJob.Pages.Add(pdf)

        ' Set MultipagePageScaling
        MyPrintJob.PrintOptions.Scaling = New MultipagePageScaling(4, 2)

        ' Print the print job
        MyPrintJob.Print()
    End Sub

    Public Shared Sub RunPageEdgeScaling()
        Dim pdf As InputPdf = New InputPdf(Util.GetPath("Resources/DocumentA.pdf"))

        ' Create a print job with the document to be printed to the default printer
        Dim MyPrintJob As PrintJob = New PrintJob(Printer.Default)
        MyPrintJob.Pages.Add(pdf)

        ' Set AutoPageScaling
        MyPrintJob.PrintOptions.Scaling = New AutoPageScaling(ScaleTo.PageEdges)

        ' Print the print job
        MyPrintJob.Print()
    End Sub

    Public Shared Sub RunPercentageScaling()
        Dim pdf As InputPdf = New InputPdf(Util.GetPath("Resources/DocumentA.pdf"))

        ' Create a print job with the document to be printed to the default printer
        Dim MyPrintJob As PrintJob = New PrintJob(Printer.Default)
        MyPrintJob.Pages.Add(pdf)

        ' Set PercentagePageScaling
        MyPrintJob.PrintOptions.Scaling = New PercentagePageScaling(0.75F)

        ' Print the print job
        MyPrintJob.Print()
    End Sub
End Class
